engine/execution/messages.go

```diff
@@ -11,38 +11,122 @@ import (
+		transactionResults:     cr.transactionResultsByCollectionIndex(colIndex),
+		executionSnapshot:      cr.StateSnapshots[colIndex],
+		serviceEvents:          cr.ServiceEvents[colIndex],
+		convertedServiceEvents: cr.ConvertedServiceEvents[colIndex],
+	}
+}
+
+// ExecutionResults captures artifacts of execution of block collections
+type ExecutionResults struct {
+	ExecutedColCounter     int
 	StateSnapshots         []*state.ExecutionSnapshot
 	Events                 []flow.EventsList
-	EventsHashes           []flow.Identifier
-	ServiceEvents          flow.EventsList
+	ServiceEvents          []flow.EventsList
+	ConvertedServiceEvents []flow.ServiceEventList
 	TransactionResults     []flow.TransactionResult
 	TransactionResultIndex []int
 
 	// TODO(patrick): switch this to execution snapshot
 	ComputationIntensities meter.MeteredComputationIntensities
-
-	ChunkDataPacks []*flow.ChunkDataPack
-	EndState       flow.StateCommitment
-
-	*execution_data.BlockExecutionData
-	*flow.ExecutionReceipt
 }
 
-func NewEmptyComputationResult(
+func NewEmptyExecutionResults(
 	block *entity.ExecutableBlock,
-) *ComputationResult {
+) *ExecutionResults {
 	numCollections := len(block.CompleteCollections) + 1
-	return &ComputationResult{
-		ExecutableBlock:        block,
+	return &ExecutionResults{
+		ExecutedColCounter:     0,
 		StateSnapshots:         make([]*state.ExecutionSnapshot, 0, numCollections),
 		Events:                 make([]flow.EventsList, numCollections),
-		EventsHashes:           make([]flow.Identifier, 0, numCollections),
-		ServiceEvents:          make(flow.EventsList, 0),
+		ServiceEvents:          make([]flow.EventsList, numCollections),
+		ConvertedServiceEvents: make([]flow.ServiceEventList, numCollections),
 		TransactionResults:     make([]flow.TransactionResult, 0),
 		TransactionResultIndex: make([]int, 0),
 		ComputationIntensities: make(meter.MeteredComputationIntensities),
-		ChunkDataPacks:         make([]*flow.ChunkDataPack, 0, numCollections),
-		EndState:               *block.StartState,
+	}
+}
+
+func (er *ExecutionResults) transactionResultsByCollectionIndex(colIndex int) []flow.TransactionResult {
+	var startTxnIndex int
+	if colIndex > 0 {
+		startTxnIndex = er.TransactionResultIndex[colIndex-1]
+	}
+	endTxnIndex := er.TransactionResultIndex[colIndex]
+	return er.TransactionResults[startTxnIndex:endTxnIndex]
+}
+
+func (er *ExecutionResults) AllServiceEvents() flow.EventsList {
+	res := make(flow.EventsList, 0)
+	for _, sv := range er.ServiceEvents {
+		if len(sv) > 0 {
+			res = append(res, sv...)
+		}
+	}
+	return res
+}
+
+func (er *ExecutionResults) AllConvertedServiceEvents() flow.ServiceEventList {
+	res := make(flow.ServiceEventList, 0)
+	for _, sv := range er.ConvertedServiceEvents {
+		if len(sv) > 0 {
+			res = append(res, sv...)
+		}
+	}
+	return res
+}
+
+// AttestationResults captures results of post-processsing execution results
+type AttestationResults struct {
```

Comment:
maybe rename to BlockAttestationResults

Reply:
renamed at [b8ccd7b](https://github.com/onflow/flow-go/pull/4078/commits/b8ccd7b4712e45aa01942322001a63e01c3b9342)